src/sfeQwiicUltrasonic.cpp

```diff
@@ -19,15 +19,18 @@ sfeTkError_t sfeQwiicUltrasonic::begin(sfeTkII2C *theBus)
         return kSTkErrFail;
 
     // Check the device address
-    if (theBus->address() < kQwiicUltrasonicMinAddress || theBus->address() > kQwiicUltrasonicMaxAddress)
+    if(_fwVersion == kQwiicUltrasonicFWOld)
     {
-        // An older version of the firmware used 0x00 as the default address.
-        // It's not really a valid address, but we need to allow it. Any other
-        // address can't be used
-        if (theBus->address() != 0x00)
-            return kSTkErrFail;
-    }
 
+        if (theBus->address() < kQwiicUltrasonicMinAddress || theBus->address() > kQwiicUltrasonicMaxAddress)
+        {
+            // An older version of the firmware used 0x00 as the default address.
+            // It's not really a valid address, but we need to allow it. Any other
+            // address can't be used
+            if (theBus->address() != 0x00)
+                return kSTkErrFail;
+        }
+    }
```

Comment:
Should also check if it's the new firmware, that the address is in the range of 0x08 to 0x77